### DISCUSSION TITLE
Is there any documentation around the Component Publishing Protocol?

### DISCUSSION THREAD

**Discussion by noahnu** [storybookjs/storybook#21382] [Documentation]
In https://storybook.js.org/docs/react/sharing/publish-storybook, it mentions the Component Publishing Protocol. I see Chromatic supports CPP level 1. If I didn't want to use Chromatic, is there an API spec for the CPP level 1 required APIs? I see "support for /metadata.json" but there's no docs on what "metadata.json" is

**Comment by itxch** [comment]
Its discussed here abit: https://github.com/storybookjs/storybook/issues/12202


But doesn't really seem well documented. I also couldn't actually get it working with chromatic which is annoying

**Comment by valentinpalkovic** [comment]
cc @kylegach, @jonniebigodes

**Comment by itxch** [comment]
@valentinpalkovic 

Over the weekend I did actually try to "reverse engineer the spec" and I did managed to get it working where:

1.  My platform creates a URL for each deployed version of storybook - and adds it to the `versions` object in the `/metadata.json` route, this file is the same across all deployed URLs.
2. Using `version=x.y.z query parameter` will take you to the right version (well actually in my case its  version=Vx.y.z)
3. There is a base URL which will display the latest version.

But when I link to it from another Storybook, composition fails - it manages to get a list of stories, and looking in network tab it fetched `/metadata.json` and `index.json` successfully, as well as the `iframe.html` page. And in the drop down I see all the versions as per `metadata.json`. But when I select another version, I get JavaScript errors. 

Again looking in the network tab its not actually calling the right URL, due to some regex issues in the Storybook code - happy to dive into here, or open a separate issue
